expression/util.go

```diff
@@ -240,3 +241,70 @@ func ConvertCol2CorCol(cond Expression, corCols []*CorrelatedColumn, outerSchema
 	}
 	return cond
 }
+
+// ConvertConstantType convert constant value to appropriate type
+// See https://dev.mysql.com/doc/refman/5.7/en/type-conversion.html for detail
+func ConvertConstantType(expr Expression) (Expression, error) {
+	scalarFunc, ok := expr.(*ScalarFunction)
+	if !ok {
+		return expr, nil
+	}
+
+	var err error
+	args := scalarFunc.GetArgs()
+
+	switch scalarFunc.FuncName.L {
+	case ast.EQ, ast.GT, ast.LE, ast.GE, ast.LT:
+		for i, arg := range args {
+			if c, ok := arg.(*Constant); ok { // check if there is a constant
+				tp := args[(i+1)%2].GetType() // only two args here
```

Comment:
I think the implementation is not easy to understand.
Can we make it more straight-forward?

The intended logic is.
```
1. check the expression is Timestamp or Datetime compare a constant.
2. if true, then convert the constant to Timestamp or Datetime.
```
But the code doesn't show this logic.